### ISSUE TITLE
Running Journaley in Wine gives truncated date numbers

### ISSUE THREAD

**Issue by tristan-k** [yyoon/Journaley#78]
I successfully installed Journaley inside a Wine-bottle and it runs really great. Except for one thing: the Date Number in the Calendar View is truncated. Can you change the size of the font or give the user a option to do so. Can you point me to the lines inside the source so I can change it myself and create a fork?

![journaley_001](https://cloud.githubusercontent.com/assets/732025/6885234/ae80e4d2-d616-11e4-864c-e0e75c357073.png)

Thanks!

**Comment by sguergachi** [comment]
You might want to take a look at the develop branch where version 2.0 is being worked on, we've made stylistic changes that might fix that problem. 

If you'd rather not have to compile the latest 2.0 build, let me know and I can post a link to the executable. 

And if you'd rather stick to the stable 1.x version then maybe if @yyoon has time he can reduce the size a little.

**Comment by tristan-k** [comment]
Thanks. Please post a link to the latest 2.0 build.  I'll report back here.

**Comment by sguergachi** [comment]
This is a link to my dropbox folder with the latest build: https://www.dropbox.com/sh/lb3tl1sl5c7c4wl/AABVN6K5vjG2U65O19e_9nLha?dl=0

I suggest you backup your journal data, 2.0 is still in development.

**Comment by tristan-k** [comment]
I just tried the latest build and the date number is still truncated. Maybe @yyoon can have look at it.

**Comment by sguergachi** [comment]
Unfortunately I'm pretty sure hes very busy at the moment, and will be at least until late April. I'm also busy at the moment.

I can tell you that the variables you'd have to play with are located in this file: https://github.com/yyoon/Journaley/blob/develop/Journaley/Controls/EntryListBox.cs

**Comment by yyoon** [comment]
Hi @tristan-k. You can tune the font sizes from the EntryListBox.cs, as @sguergachi pointed. Would you be able to build the project yourself?

**Comment by tristan-k** [comment]
To be honest I've never build in Windows before. I'm familiar with the build process in Unix. I would appreciate your effort.

**Comment by sguergachi** [comment]
As much as I'd love for you to use Journaley, I don't know if it's worth the hassle to get it running on Linux. Even if this font size issue is resolved, I'm sure there will be other platform related issues that aren't so easily solvable you might run into down the road. And because we officially support Windows only, we don't plan to invest a lot of time to make sure the app doesn't break in Wine. 

I see that you've stared the DayOneViewer repo, which is written in Java. Have you had success with that?

**Comment by tristan-k** [comment]
DayOneViewer is nowhere near as comfortable as Journaley. It lacks a decent calendar view. Sadly there are no other gui options available for linux users.

**Comment by yyoon** [comment]
Hi @tristan-k,
Can you try this one? I have included multiple exe files built with different day font sizes, using the latest dev version.
https://dl.dropboxusercontent.com/u/10557104/Journaley-SmallerDayFonts.zip

**Comment by tristan-k** [comment]
I'm unable to run this version. It immediatley crashes. 

```
Unhandled exception: 0xe0434352 in 32-bit code (0x7b83b6ae).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:7b83b6ae ESP:0032f024 EBP:0032f098 EFLAGS:00000287(   - --  I S - -P-C)
 EAX:7b827121 EBX:7b8bc000 ECX:79140000 EDX:0032f058
 ESI:e0434352 EDI:00143908
Stack dump:
0x0032f024:  0032f0f8 00000014 7923b348 e0434352
0x0032f034:  00000001 00000000 7b83b6ae 00000005
0x0032f044:  80131539 00000000 00000000 00000000
0x0032f054:  79140000 00000000 0032f0f8 003624b0
0x0032f064:  02000059 0032f078 79150579 0032f080
0x0032f074:  02000059 0032f084 7915bc5e 003dc828
000c: sel=0067 base=00000000 limit=00000000 32-bit r-x
Backtrace:
=>0 0x7b83b6ae in kernel32 (+0x2b6ae) (0x0032f098)
  1 0x791cac08 in clr (+0x8ac07) (0x0032f12c)
  2 0x791cae41 in clr (+0x8ae40) (0x0032f1ec)
  3 0x0583d268 (0x0032f1f8)
  4 0x0583d1dd (0x0032f224)
  5 0x0583d03f (0x0032f250)
  6 0x03807c8e (0x0032f298)
  7 0x03802735 (0x0032f2b4)
  8 0x791421db in clr (+0x21da) (0x0032f2c4)
  9 0x79164a2a in clr (+0x24a29) (0x0032f340)
  10 0x79164bcc in clr (+0x24bcb) (0x0032f480)
  11 0x79164c01 in clr (+0x24c00) (0x0032f49c)
  12 0x79164c21 in clr (+0x24c20) (0x0032f4b4)
  13 0x7922ce82 in clr (+0xece81) (0x0032f618)
  14 0x7922cf90 in clr (+0xecf8f) (0x0032f880)
  15 0x7922cda4 in clr (+0xecda3) (0x0032fd64)
  16 0x7922d199 in clr (+0xed198) (0x0032fdb8)
  17 0x7922d09a in clr (+0xed099) (0x0032fe04)
  18 0x792aaf00 in clr (+0x16aeff) (0x0032fe3c)
  19 0x603b55ab in mscoreei (+0x55aa) (0x0032fe48)
  20 0x79007f16 in mscoree (+0x7f15) (0x0032fe58)
  21 0x79004de3 in mscoree (+0x4de2) (0x0032fe78)
  22 0x7b8617b3 in kernel32 (+0x517b2) (0x0032feb8)
  23 0x7bc80fb0 call_thread_func_wrapper+0xb() in ntdll (0x0032fed8)
  24 0x7bc8417d call_thread_func+0x7c() in ntdll (0x0032ffa8)
  25 0x7bc80f8e RtlRaiseException+0x21() in ntdll (0x0032ffc8)
  26 0x7bc54b9e call_dll_entry_point+0x3fd() in ntdll (0x0032ffe8)
  27 0xf752e65d wine_call_on_stack+0x1c() in libwine.so.1 (0x00000000)
  28 0xf752e71b wine_switch_to_stack+0x2a() in libwine.so.1 (0xff7fc838)
  29 0x7bc5a709 LdrInitializeThunk+0x238() in ntdll (0xff7fc878)
  30 0x7b868043 __wine_kernel_init+0xa12() in kernel32 (0xff7fd998)
  31 0x7bc5b633 __wine_process_init+0x192() in ntdll (0xff7fda28)
  32 0xf752bdc8 wine_init+0x327() in libwine.so.1 (0xff7fda88)
  33 0x7bf0100c main+0xfb() in <wine-loader> (0xff7fded8)
  34 0xf734ba83 __libc_start_main+0xf2() in libc.so.6 (0x00000000)
0x7b83b6ae: subl	$4,%esp
Modules:
Module	Address			Debug info	Name (143 modules)
PE	  400000-  514000	Deferred        journaley-dayfont17
PE	55000000-5519a000	Deferred        wpfgfx_v0400
PE	554c0000-55585000	Deferred        presentationnative_v0400
PE	561c0000-562a1000	Deferred        wpftxt_v0400
PE	5e0d0000-5e17a000	Deferred        diasymreader
PE	60340000-6034d000	Deferred        culture
PE	603b0000-60416000	Export          mscoreei
PE	60930000-60940000	Deferred        nlssorting
PE	714b0000-71576000	Deferred        naturallanguage6
PE	79000000-7904a000	Export          mscoree
PE	79060000-7911e000	Deferred        msvcr100_clr0400
PE	79140000-797af000	Export          clr
PE	79810000-79870000	Deferred        clrjit
ELF	7a800000-7a91e000	Deferred        opengl32<elf>
  \-PE	7a820000-7a91e000	\               opengl32
ELF	7b800000-7ba62000	Dwarf           kernel32<elf>
  \-PE	7b810000-7ba62000	\               kernel32
ELF	7bc00000-7bce7000	Dwarf           ntdll<elf>
  \-PE	7bc10000-7bce7000	\               ntdll
ELF	7bf00000-7bf04000	Dwarf           <wine-loader>
ELF	7c915000-7c94c000	Deferred        libtxc_dxtn.so
ELF	7c96f000-7c98c000	Deferred        libgcc_s.so.1
ELF	7ca75000-7ca83000	Deferred        libdrm_radeon.so.1
ELF	7ca83000-7cfdc000	Deferred        i965_dri.so
ELF	7cfdc000-7d027000	Deferred        libdbus-1.so.3
ELF	7d027000-7d0b9000	Deferred        libgl.so.1
ELF	7d0b9000-7d200000	Deferred        wined3d<elf>
  \-PE	7d0d0000-7d200000	\               wined3d
ELF	7d302000-7d30d000	Deferred        libpciaccess.so.0
ELF	7d30d000-7d315000	Deferred        libdrm_nouveau.so.2
ELF	7d315000-7d337000	Deferred        libdrm_intel.so.1
ELF	7d337000-7d340000	Deferred        librt.so.1
ELF	7d340000-7d34a000	Deferred        libnih-dbus.so.1
ELF	7d34a000-7d368000	Deferred        libcgmanager.so.0
ELF	7d428000-7d441000	Deferred        libnih.so.1
ELF	7d441000-7d454000	Deferred        libudev.so.1
ELF	7d454000-7d461000	Deferred        libdrm.so.2
ELF	7d461000-7d468000	Deferred        libxcb-sync.so.1
ELF	7d468000-7d46c000	Deferred        libxcb-present.so.0
ELF	7d46c000-7d484000	Deferred        libxcb-glx.so.0
ELF	7d484000-7d49d000	Deferred        libglapi.so.0
ELF	7d4c0000-7d4fe000	Deferred        d3d9<elf>
  \-PE	7d4d0000-7d4fe000	\               d3d9
ELF	7d4fe000-7d52e000	Deferred        p11-kit-trust.so
ELF	7d52e000-7d535000	Deferred        libffi.so.6
ELF	7d535000-7d571000	Deferred        libp11-kit.so.0
ELF	7d571000-7d585000	Deferred        libtasn1.so.6
ELF	7d585000-7d60c000	Deferred        libgcrypt.so.11
ELF	7d60c000-7d6d2000	Deferred        libgnutls.so.26
ELF	7d6d2000-7d6d5000	Deferred        libxshmfence.so.1
ELF	7d6d5000-7d6d9000	Deferred        libxcb-dri3.so.0
ELF	7d6d9000-7d6df000	Deferred        libxcb-dri2.so.0
ELF	7d6df000-7d6e2000	Deferred        libx11-xcb.so.1
ELF	7d6e2000-7d6f5000	Deferred        gnome-keyring-pkcs11.so
ELF	7d6f5000-7d71c000	Deferred        iphlpapi<elf>
  \-PE	7d700000-7d71c000	\               iphlpapi
ELF	7d71c000-7d74b000	Deferred        netapi32<elf>
  \-PE	7d720000-7d74b000	\               netapi32
ELF	7d74b000-7d77e000	Deferred        secur32<elf>
  \-PE	7d750000-7d77e000	\               secur32
ELF	7d77e000-7d798000	Deferred        sxs<elf>
  \-PE	7d780000-7d798000	\               sxs
ELF	7d798000-7d7d1000	Deferred        ws2_32<elf>
  \-PE	7d7a0000-7d7d1000	\               ws2_32
ELF	7d7d1000-7d7f9000	Deferred        mpr<elf>
  \-PE	7d7e0000-7d7f9000	\               mpr
ELF	7d7f9000-7d876000	Deferred        wininet<elf>
  \-PE	7d800000-7d876000	\               wininet
ELF	7d876000-7d919000	Deferred        urlmon<elf>
  \-PE	7d880000-7d919000	\               urlmon
ELF	7d919000-7d9aa000	Deferred        ieframe<elf>
  \-PE	7d920000-7d9aa000	\               ieframe
ELF	7d9aa000-7dbe1000	Deferred        shell32<elf>
  \-PE	7d9c0000-7dbe1000	\               shell32
ELF	7dbe1000-7dc92000	Deferred        msvcrt<elf>
  \-PE	7dc00000-7dc92000	\               msvcrt
ELF	7dc92000-7dd51000	Deferred        windowscodecs<elf>
  \-PE	7dca0000-7dd51000	\               windowscodecs
ELF	7dd51000-7dddc000	Deferred        gdiplus<elf>
  \-PE	7dd60000-7dddc000	\               gdiplus
ELF	7dddc000-7df21000	Deferred        oleaut32<elf>
  \-PE	7ddf0000-7df21000	\               oleaut32
ELF	7df21000-7e02c000	Deferred        comctl32<elf>
  \-PE	7df30000-7e02c000	\               comctl32
ELF	7e02c000-7e0fc000	Deferred        crypt32<elf>
  \-PE	7e040000-7e0fc000	\               crypt32
ELF	7e0fc000-7e13f000	Deferred        rsaenh<elf>
  \-PE	7e100000-7e13f000	\               rsaenh
ELF	7e13f000-7e176000	Deferred        uxtheme<elf>
  \-PE	7e150000-7e176000	\               uxtheme
ELF	7e1bc000-7e1c2000	Deferred        libxfixes.so.3
ELF	7e1c2000-7e1cd000	Deferred        libxcursor.so.1
ELF	7e1cd000-7e1de000	Deferred        libxi.so.6
ELF	7e1de000-7e1e2000	Deferred        libxcomposite.so.1
ELF	7e1e2000-7e1ed000	Deferred        libxrandr.so.2
ELF	7e1ed000-7e1f8000	Deferred        libxrender.so.1
ELF	7e1f8000-7e1fe000	Deferred        libxxf86vm.so.1
ELF	7e1fe000-7e202000	Deferred        libxinerama.so.1
ELF	7e202000-7e209000	Deferred        libxdmcp.so.6
ELF	7e209000-7e20d000	Deferred        libxau.so.6
ELF	7e20d000-7e22f000	Deferred        libxcb.so.1
ELF	7e22f000-7e363000	Deferred        libx11.so.6
ELF	7e363000-7e376000	Deferred        libxext.so.6
ELF	7e378000-7e37c000	Deferred        libxdamage.so.1
ELF	7e37c000-7e381000	Deferred        libgpg-error.so.0
ELF	7e381000-7e397000	Deferred        dwmapi<elf>
  \-PE	7e390000-7e397000	\               dwmapi
ELF	7e399000-7e42d000	Deferred        winex11<elf>
  \-PE	7e3a0000-7e42d000	\               winex11
ELF	7e42d000-7e4b1000	Deferred        rpcrt4<elf>
  \-PE	7e440000-7e4b1000	\               rpcrt4
ELF	7e4b1000-7e5f4000	Deferred        ole32<elf>
  \-PE	7e4d0000-7e5f4000	\               ole32
ELF	7e5f4000-7e619000	Deferred        imm32<elf>
  \-PE	7e600000-7e619000	\               imm32
ELF	7e6ae000-7e6d7000	Deferred        libexpat.so.1
ELF	7e6d7000-7e712000	Deferred        libfontconfig.so.1
ELF	7e712000-7e73a000	Deferred        libpng12.so.0
ELF	7e73a000-7e754000	Deferred        libz.so.1
ELF	7e754000-7e7f4000	Deferred        libfreetype.so.6
ELF	7e817000-7e936000	Deferred        gdi32<elf>
  \-PE	7e820000-7e936000	\               gdi32
ELF	7e936000-7ea92000	Deferred        user32<elf>
  \-PE	7e950000-7ea92000	\               user32
ELF	7ea92000-7eb0c000	Deferred        shlwapi<elf>
  \-PE	7eaa0000-7eb0c000	\               shlwapi
ELF	7eb32000-7ebad000	Deferred        advapi32<elf>
  \-PE	7eb40000-7ebad000	\               advapi32
ELF	7ebad000-7ebcf000	Deferred        libtinfo.so.5
ELF	7ebcf000-7ebf4000	Deferred        libncurses.so.5
ELF	7ef65000-7ef72000	Deferred        libnss_files.so.2
ELF	7ef72000-7ef7e000	Deferred        libnss_nis.so.2
ELF	7ef7e000-7ef97000	Deferred        libnsl.so.1
ELF	7ef97000-7efdd000	Deferred        libm.so.6
ELF	7efdf000-7eff9000	Deferred        version<elf>
  \-PE	7efe0000-7eff9000	\               version
ELF	f7332000-f74e0000	Dwarf           libc.so.6
ELF	f74e0000-f74e5000	Deferred        libdl.so.2
ELF	f74e6000-f7502000	Deferred        libpthread.so.0
ELF	f7507000-f7510000	Deferred        libnss_compat.so.2
ELF	f7525000-f76db000	Dwarf           libwine.so.1
ELF	f76dd000-f76ff000	Deferred        ld-linux.so.2
ELF	f7701000-f7702000	Deferred        [vdso].so
Threads:
process  tid      prio (all id:s are in hex)
00000008 (D) Z:\home\tristank\Downloads\Journaley\Journaley-DayFont17.exe
	0000002a    0
	00000027    2
	00000026    0
	00000009    0 <==
0000000e services.exe
	00000031    0
	00000030    0
	0000002f    0
	00000022    0
	0000001b    0
	00000016    0
	00000010    0
	0000000f    0
00000019 winedevice.exe
	00000021    0
	0000001e    0
	0000001d    0
	0000001a    0
0000001f plugplay.exe
	00000025    0
	00000024    0
	00000020    0
00000028 explorer.exe
	00000029    0
0000002d WPFFontCache_v0400.exe
	00000037    0
	00000036    0
	00000035    0
	00000034    0
	00000033    0
	00000032    0
	0000002e    0
0000003a wineconsole.exe
	0000003b    0
System information:
    Wine build: wine-1.7.38
    Platform: i386
    Host system: Linux
    Host version: 3.18.0-ctx.patch

```

**Comment by yyoon** [comment]
Hmm. Could you try this one instead? These are based off of the 1.4 release.
https://dl.dropboxusercontent.com/u/10557104/Journaley1.4-SmallerDayFonts.zip

**Comment by tristan-k** [comment]
That worked. Thank you very much. 

![journaley_001](https://cloud.githubusercontent.com/assets/732025/6996795/b1f59cce-db9e-11e4-8fbc-bac2593e36ab.png)

**Comment by tristan-k** [comment]
I dont want to be a prick, but can you built another version with Fontsize 14 and 17? Size 20 is still to big for the actual time. Sorry. 

![journaley_004](https://cloud.githubusercontent.com/assets/732025/7101207/eb15e2c2-e04c-11e4-9c16-793bef087a67.png)

**Comment by yyoon** [comment]
The date and time are using different font sizes.
Maybe try one of these:
https://dl.dropboxusercontent.com/u/10557104/Journaley1.4-SmallerTimeFonts.zip

**Comment by yyoon** [comment]
This should be fixed by 9d66859ac2b715b0b15db918918b6a6557d81edd.

In summary, Journaley measures the width of the day / time text first, and if the measured width value exceeds the allotted size, it decreases the font size dynamically.
